ts/parts-map/HeatmapSeries.ts

```diff
@@ -514,7 +591,49 @@ seriesType<Highcharts.HeatmapSeriesOptions>(
 
         /* eslint-enable valid-jsdoc */
 
-    }, colorPointMixin)
+    },
+    colorPointMixin,
+    {
+        setState: function (
+            this: Highcharts.HeatmapPoint,
+            state?: string
+        ): void {
+            var point = this,
+                series = this.series,
+                points = series.points,
+                inactiveException = series.options.states &&
+                        series.options.states.inactive &&
+                        series.options.states.inactive.except,
+                filterFunction,
+                i = 0;
+
+            if (inactiveException) {
```

Comment:
Would it not be better to keep this logic in [`applyInactiveState`](https://github.com/highcharts/highcharts/blob/3afea52337b352c0f4f848312f3425c2105bc38e/ts/parts/Pointer.ts#L873), and keep `setState` as before?

To me it seems like `Point.setState` will now, maybe unexpectedly, alter the state of many points, instead of just the one `setState` was called on.

Reply:
Good point. Maybe instead of `pointer.applyInactiveState` the [`series.setAllPointsToState`](https://github.com/highcharts/highcharts/blob/3afea52337b352c0f4f848312f3425c2105bc38e/ts/parts/Interaction.ts#L1546-L1555) would we a better place.

Logic behind using `Point.setState` was inherited from networkgraph, where one point's state propagates to all connected nodes.

Reply:
> the series.setAllPointsToState would we a better place

Sounds good to me.

> Logic behind using Point.setState was inherited from networkgraph, where one point's state propagates to all connected nodes.

I see, I have the same worry for this.
Though if it is fine to alter other points in `setState`, then maybe the `Series.setState` could be extended to call an `applyStateToOtherPoints` or something, if it exists on the prototype?
Then all series could just implement this, instead of re-implementing `setState`. 🤔

Reply:
In a separate PR I will try to refactor `networkgraph` (and other node-based series) to use `series.setAllPointsToState` instead of `setState`. So the solution will be consistent across series.

I just realized one thing: `setState()` is part of our API. Maybe we should ask ourselves  a question: should `point.setState()`, called by user, alter other point's state? I think this might be considered as a bug, so I would rather make `setAllPointsToState` available in API (if someone request that). What do you think?

Reply:
I agree to this.

> Maybe we should ask ourselves a question: should point.setState(), called by user, alter other points' state?

Networkgraph has been working around this issue by having a [third undocumented variable](https://github.com/highcharts/highcharts/blob/d18eaabe4749dff19afac6f66f7ccdd4c13d973d/ts/modules/networkgraph/networkgraph.src.ts#L985), but I believe what you describe is a cleaner solution.

Reply:
Just to be on the same page:
The third argument, is actually the second argument and it's [documented](https://api.highcharts.com/class-reference/Highcharts.Series#setState) - note it's `series.setState()`, not `point.setState()`.

Reply:
Yes, you are right. The Networkgraph `Point.setState` is defined at [mixins/nodes.ts#L283-L309](https://github.com/highcharts/highcharts/blob/d18eaabe4749dff19afac6f66f7ccdd4c13d973d/ts/mixins/nodes.ts#L283-L309), and it seems that altering of other point states can not be toggled as it is now.

Reply:
It's refactored now, no more `setNodeState` method, everything is done using `inactiveFilters` in all node-based series.